src/huggingface_hub/inference/_client.py

```diff
@@ -252,6 +248,14 @@ def post(
         stream: bool = False,
     ) -> Union[bytes, Iterable[bytes]]: ...
 
+    @_deprecate_method(
+        version="0.31.0",
+        message=(
+            "Making direct POST requests to the inference server is not supported anymore. "
+            "Please use task methods instead (e.g. `InferenceClient.chat_completion`). "
+            "If your use case is not supported, please open an issue in https://github.com/huggingface/huggingface_hub."
+        ),
+    )
```

Comment:
Shouldn't have made this one public :confused: (my bad)
Will rely on `_inner_post` from now on which is private so we can change everything.